semgrep-core/src/core/Xtarget.ml

```diff
@@ -27,5 +27,5 @@ type t = {
   lazy_content : string lazy_t;
   (* This is valid only for xlang = Xlang.L ..., not for LRegex|LGeneric *)
   lazy_ast_and_errors :
-    (AST_generic.program * Semgrep_error_code.error list) lazy_t;
+    (AST_generic.program * Parse_info.token_location list) lazy_t;
```

Comment:
Don't we want to allow for any kind of error here? We could always filter the ones that are interesting for the CLI later. I am trying to update DeepSemgrep and we have:

```
    try
    let { Parse_target.ast; skipped_tokens; _} = 
          Parse_target.parse_and_resolve_name
          lang file in
    file, (ast, skipped_tokens)
    with
    | exn -> file, ([], [exn_to_error file exn])
```
What to do with that `exn` then?

Reply:
ok, I'll try to update deep-semgrep and see if we need to go revert some code in semgrep.

Reply:
Here the only errors we were getting were always skipped_tokens, so I preferred to be precise.
The parsing function can also return exns (e.g., ParseError, LexicalError, etc.), but once we get an AST,
the only possible thing attached to it are skipped_tokens.